src/logic/azext_logic/generated/action.py

```diff
@@ -0,0 +1,67 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=protected-access
+
+import argparse
+from knack.util import CLIError
+
+
+class AddIntegrationAccount(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        action = self.get_action(values, option_string)
+        namespace.integration_account = action
+
+    def get_action(self, values, option_string):  # pylint: disable=no-self-use
+        try:
+            properties = dict(x.split('=', 1) for x in values)
+        except ValueError:
+            raise CLIError('usage error: {} [KEY=VALUE ...]'.format(option_string))
+        d = {}
+        for k in properties:
+            kl = k.lower()
+            v = properties[k]
+            if kl == 'id':
+                d['id'] = v
+        return d
+
+
+class AddSku(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
```

Comment:
This is auto-generated code via autorest, no custom code in this file. Is your suggestion here for me to request the codegen team changes this then? I could of course manually change this, but then we will just run into this problem in the future with other entities with only one properties

Reply:
@RodgeFu

Reply:
@RodgeFu  you can try to config to further flatten the sku instead of manual code if that's what wanted

Would this change this code as well?

Reply:
yes

Reply:
@RodgeFu why this is not flattened by default, if an object only has 1 or 2 properties, I believe previous behavior is to flatten them by default.

Reply:
I see that the `--sku` is an array, that might be the reason why action is generated by default. But if the object only has 1 property, even with array we should still flatten it and use the parameter like `--sku sku1 sku2`. In this particular case, it's weird to have an array for sku. Probably a swagger issue? Please double check.

Reply:
i think now action is used if possible instead of flatten, i agree that flatten should be used if it only has 1 property. Please file an issue for it. Could you double confirm whether it's preferred in CLI to use flatten for 2 properties? because we have the same 2 threshold in python SDK and found it would cause many backward compatible problem when service team add more properties into the schema. 

@bquantump, before the change, i think you can use directive to config flatten on the sku to get what you want here. you can find more details about how to config it at https://github.com/Azure/autorest.az/blob/master/doc/faq.md

Reply:
the integration account sku is not an array.

https://github.com/Azure/azure-rest-api-specs/blob/e61d6ea2ea4c355f8ae344ad964690107e377c99/specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json#L9481

https://github.com/Azure/azure-rest-api-specs/blob/e61d6ea2ea4c355f8ae344ad964690107e377c99/specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json#L9156
 
You are right in that the codegen put it as nargs='+'. So, I think this is a codegen issue. It should just model this as string, as the swagger says to do, and that's that. 

 (Also the workflow sku is read-only so they is irrelevant to this since the user cannot set this for a workflow. )

Reply:
Based on the swagger, sku should not have `nargs='+'`. Please submit an issue in autorest.az repo. You can fix it manually to unblock this PR.

Reply:
@RodgeFu In the case of array, we cannot use flatten feature if the object has 2 properties or more. So it's fine to default to use action. I was not aware that the action is only generated for array property in my first comment.

Reply:
Fixed manually, ticket is filed here: https://github.com/Azure/autorest.az/issues/336